PD-1817 / None / Pd 1817 master message kubernetes migration and 24 10 apps update cutoff (#3577)
truenas/documentation: static/includes/AppsHostIP.md

&NewLine;

Application host IP port binding is being developed for all applications in the TrueNAS Apps catalog, starting with TrueNAS 25.04.
This feature allows per-app selection of any IP address from the available aliases assigned to an interface to bind the WebUI port to.
It includes port bind mode options to publish the port for external access or expose it for inter-container communication.

A small but growing list of applications currently support this functionality in TrueNAS 24.10 or later.

{{< expand "Applications that currently support host IP port binding" "v" >}}
The following applications currently support host IP port binding.
{{< columns >}}
* calibre-web  
* esphome  
* handbrake-web  
* homearr  
* invoiceninja  
* it-tools  
* jelu  
* lyrion-music-server  
* minecraft-bedrock  

<--->

* romm  
* satisfactory-server  
* steam-headless  
* terreria  
* tianji  
* umami  
* urbackup  
* zigbee2mqtt  
* emby  
{{< /columns >}}
All future applications, as well as those added to the TrueNAS Apps catalog after December 24, 2024, support this feature.
{{< /expand >}}

> However, applications that were in the TrueNAS Apps catalog before implementation of this feature require OS-level changes to enable support.

Reviewer: "However" seems out of place given the preceding sentence, which I assume is why you used "however", is inside the expand. Can you reword this without using it?

Author: Take the expand out of it, the preceding sentence is 

> A small but growing list of applications currently support this functionality in TrueNAS 24.10 or later.
...
> However, applications that were in the TrueNAS Apps catalog before implementation of this feature require OS-level changes to enable support.

It reads better rendered than in the markdown.

As a result, June 1 is also the cutoff date for two related app behaviors:

* **24.04 to 24.10 App Migrations:**
  TrueNAS 24.10 introduced a new Docker-based TrueNAS Apps backend and automated migration for Kubernetes-based apps on upgrade.

  Due to breaking changes involved in enabling host IP port binding, **June 1, 2025** is the deadline for automatic apps migration on upgrade.

>   Any users still running TrueNAS Apps on 24.04 after June 1 will have to re-deploy those apps after upgrading to 24.10 or later.

Reviewer: do not use "will"
```suggestion
  Any users still running TrueNAS Apps on 24.04 after June 1 must re-deploy those apps after upgrading to 24.10 or later.
```

* **24.10 App Updates:**
  The cutoff date for app updates on TrueNAS 24.10 is **June 1, 2025**.

>   Any users still running TrueNAS Apps on 24.10 after June 1 will have to update TrueNAS to 25.04 (or later) to update or install new applications.

Reviewer: don't use "will"
```suggestion
  Any users still running TrueNAS Apps on 24.10 after June 1 must update TrueNAS to 25.04 (or later) to update or install new applications.
```

Author: Same comment
